core/allow_insecure_certificates_deserializer.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use std::fmt::{self, Formatter};
+use std::marker::PhantomData;
+
+use serde::{de, Deserialize, Deserializer};
+
+pub fn deserialize_allow_insecure_certificates<'de, D>(
+  deserializer: D,
+) -> Result<Option<Vec<String>>, D::Error>
+where
+  D: Deserializer<'de>,
+{
+  struct OptStringOrBoolOrVec(PhantomData<Option<Vec<String>>>);
+
+  impl<'de> de::Visitor<'de> for OptStringOrBoolOrVec {
+    type Value = Option<Vec<String>>;
+
+    fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
+      formatter.write_str("a string or sequence of strings")
+    }
+
```

Comment:
I don't understand why this custom deserializer is needed. Why can't we just use `Option<Vec<String>>` in struct fields?

Reply:
This custom deserializer converts a boolean value, a string or a sequence of strings to the `Option<Vec<String>>`. The boolean value when `true` - sets the `Option` to `Some(vec![])` or when `false` - to `None`. The `allow_insecure_certificates` matches the CLI behavior which can be a `boolean` (like a CLI flag `--allow-insecure-certificates`) or a value (like in CLI `--allow-insecure-certificates=domain.com`).

What should be accepted as an input when using `connectTls` or `createHttpClient` to exclude the validation for specific domains - or should be only defined globally through  the CLI flag?

Reply:
> What should be accepted as an input when using connectTls or createHttpClient to exclude the validation for specific domains - or should be only defined globally through the CLI flag?

I discussed this with @ry and we believe this should only be configured via CLI flag.